src/main/java/org/terasology/taskSystem/actions/PerformTaskNode.java

```diff
@@ -215,32 +216,28 @@ private void completeTask(Actor oreon, TaskComponent taskComponent) {
                 if (buildingTemplate != null) {
                     List<SpawnBlockRegionsComponent.RegionToFill> relativeRegionsToFill = buildingTemplate.getParentPrefab().getComponent(SpawnBlockRegionsComponent.class).regionsToFill;
 
-                    List<Region3i> regionsToFill = new ArrayList<>();
+                    List<BlockRegion> regionsToFill = new ArrayList<>();
                     for (SpawnBlockRegionsComponent.RegionToFill regionToFill : relativeRegionsToFill) {
                         BlockRegion relativeRegion = regionToFill.region;
-                        BlockRegion absoluteRegion = relativeRegion.translate(JomlUtil.from(buildingComponent.centerLocation));
-                        regionsToFill.add(JomlUtil.from(absoluteRegion));
+                        BlockRegion absoluteRegion = relativeRegion.translate(buildingComponent.centerLocation, new BlockRegion(BlockRegion.INVALID));
+                        regionsToFill.add(absoluteRegion);
                     }
-                    Region3i totalRegion = Region3i.createFromMinAndSize(new Vector3i(regionsToFill.get(0).center().x, taskManagementSystem.minYOverall, regionsToFill.get(0).center().z), new Vector3i(1, 1, 1));
-                    for (Region3i baseRegion:regionsToFill) {
+                    Vector3i center = new Vector3i(regionsToFill.get(0).center(new Vector3f()), RoundingMode.FLOOR);
+                    BlockRegion totalRegion = new BlockRegion(center.x, taskManagementSystem.minYOverall, center.z).setSize(1, 1, 1);
+                    for (BlockRegion baseRegion : regionsToFill) {
 
-                        Vector3i min = new Vector3i(baseRegion.minX(), taskManagementSystem.minYOverall+baseRegion.minY(), baseRegion.minZ());
-                        Vector3i max = new Vector3i(baseRegion.maxX(), baseRegion.maxY()+taskManagementSystem.minYOverall, baseRegion.maxZ());
+                        Vector3i min = new Vector3i(baseRegion.minX(), taskManagementSystem.minYOverall + baseRegion.minY(), baseRegion.minZ());
+                        Vector3i max = new Vector3i(baseRegion.maxX(), baseRegion.maxY() + taskManagementSystem.minYOverall, baseRegion.maxZ());
 
-                        Region3i region = Region3i.createFromMinMax(min, max);
-                        Iterator<Vector3i> regionsIterator = region.iterator();
-                        while (regionsIterator.hasNext()) {
-                            Vector3i vector = regionsIterator.next();
-                            totalRegion = totalRegion.expandToContain(vector);
+                        BlockRegion region = new BlockRegion(min, max);
+                        for (Vector3ic v : region) {
+                            totalRegion.union(v);
```

Comment:
This is basically the same as building the union of `totalRegion.union(region)` which in turn is the same as `totalRegion.union(min).union(max)` without the need to create another block region. We could even get away without vector allocations...